### ISSUE TITLE
X-Varnish-Hits stays at 0

### ISSUE THREAD

**Issue by mrgray** [cscart/full-page-cache-addon#4]
Haven't modified the schema, deleted all cookies for the domain and X-Varnish-Hits still stays at 0 for the homepage, static pages and all product pages.

**Comment by alpharder** [comment]
Please consider sharing more details when posting an issue so we could have more information to start with. What are the other headers?

**Comment by gongled** [comment]
@mrgray @alpharder I've reproduced this issue on CS-Cart 4.3.9 and full-page-cache-addon 0.0.8. I suppose it might be connected to User-Agent header. I prepared some tests and found that request without this header is not cached in Varnish.

### Environment

CentOS 6 x86_64, NGINX 1.11.3, Varnish 4.0.3, Apache 2.2.15, PHP 5.6.24.

### Steps to reproduce

Please run the commands in the following order:

#### Without User-Agent header:
```
$ curl -sI http://example.com/video-games/ -H 'Connection: keep-alive' \
 -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8' \
 -H 'Accept-Encoding: gzip, deflate, sdch' \
 -H 'Accept-Language: en-US,en;q=0.8' \
 -H 'Cookie: sid_customer_a9b9f=927eb66f50ccd9c6ec07be1970ac49e3-1-C' | grep X-Varnish-Hits
X-Varnish-Hits: 2

$ curl -sI http://example.com/video-games/ -H 'Connection: keep-alive' \
 -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8' \
 -H 'Accept-Encoding: gzip, deflate, sdch' \
 -H 'Accept-Language: en-US,en;q=0.8' \
 -H 'Cookie: sid_customer_a9b9f=927eb66f50ccd9c6ec07be1970ac49e3-1-C' | grep X-Varnish-Hits
X-Varnish-Hits: 3
```

#### With User-Agent header:
```
$ curl -sI http://example.com/video-games/ -H 'Connection: keep-alive' \
 -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/54.0.2833.0 Safari/537.36' \
 -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8' \
 -H 'Accept-Encoding: gzip, deflate, sdch' \
 -H 'Accept-Language: en-US,en;q=0.8' \
 -H 'Cookie: sid_customer_a9b9f=927eb66f50ccd9c6ec07be1970ac49e3-1-C' | grep X-Varnish-Hits
X-Varnish-Hits: 0

$ ~/Downloads/full-page-cache-addon-dev/app/addons/full_page_cache → curl -sI http://example.com/video-games/ -H 'Connection: keep-alive' \
 -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/54.0.2833.0 Safari/537.36' \
 -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8' \
 -H 'Accept-Encoding: gzip, deflate, sdch' \
 -H 'Accept-Language: en-US,en;q=0.8' \
 -H 'Cookie: sid_customer_a9b9f=927eb66f50ccd9c6ec07be1970ac49e3-1-C' | grep X-Varnish-Hits
X-Varnish-Hits: 0```

Please let me know if you need detailed information about host configuration or your have any questions.

**Comment by alpharder** [comment]
@gongled Have you mixed up the "Without User-Agent header:" and "With User-Agent header:" sentences?

**Comment by mrgray** [comment]
Sorry for not replying to this. I wasn't subscribed to the thread for some reason. 

I ended up not using varnish because of a bunch of reasons,

- CS-Cart version was 4.3.6 
- Varnish had to be manually restarted whenever code got pushed to the server to reset the cache (might just have been my poorly configured caching defaults)
- I did not realize that the default config basically had no caching enabled by default (should probably document this)
- Was using localizations comprehensively; possibly an issue?
- We actually moved from a LiteSpeed to Apache server. Way too bumpy in production, so we relegated to APC + Cloudflare.

**Comment by mrgray** [comment]
Here's some more information, after I had opened this ticket, I redeployed Varnish's configuration with a lot of back and forth with my hosting provider, I had managed to make caching work for static, category and product pages - so it **does work** - assuming you can fine-tune the config _just right_. 

But I did not trust the config enough in production because I was unsure of how well it would handle the localized currencies for cached pages.

**Comment by gongled** [comment]
@alpharder yes, I did it one by one in the same order as I specified before.

**Comment by gongled** [comment]
@mrgray sorry for the off-topic, but why did you moved from a LiteSpeed to Apache? I've never used LiteSpeed before, so I'm very curious to hear your story of using it in production.

**Comment by alpharder** [comment]
This is weird because the User-Agent header shouldn't be taken into account. But okay now we have steps to reproduces.
